### ISSUE TITLE
SamplerResult.quasi_dists should return QuasiDistributions

### ISSUE THREAD

**Issue by kevinsung** [Qiskit/qiskit-ibm-runtime#493]
**Describe the bug**
Instead, it returns `dicts`.

**Steps to reproduce**
```python
from qiskit import QuantumCircuit
from qiskit.primitives import Sampler as QuantumInfoSampler
from qiskit_ibm_runtime import Sampler as RuntimeSampler

bell = QuantumCircuit(2)
bell.h(0)
bell.cx(0, 1)
bell.measure_all()

sampler = QuantumInfoSampler()
job = sampler.run([bell])
result = job.result()
quasi_dists = result.quasi_dists
print(type(quasi_dists[0]))

sampler = RuntimeSampler(options={"backend": "ibmq_qasm_simulator"})
job = sampler.run([bell])
result = job.result()
quasi_dists = result.quasi_dists
print(type(quasi_dists[0]))
```
```
<class 'qiskit.result.distributions.quasi.QuasiDistribution'>
<class 'dict'>
```

**Expected behavior**
See title.

**Suggested solutions**
See title.

**Additional Information**

- **qiskit-ibm-runtime version**: 59b715e6769d1307045a6f042ca584cd47e4ea5c
- **Python version**: 3.9.13
- **Operating system**: Arch Linux

**Comment by kevinsung** [comment]
I just noticed that this is a duplicate of https://github.com/Qiskit/qiskit-ibm-runtime/issues/362.

**Comment by jyu00** [comment]
Closing since it's a duplicate.

**Comment by kevinsung** [comment]
It's a duplicate, but the duplicate got closed even though this won't actually be resolved until https://github.com/Qiskit/qiskit-ibm-runtime/pull/511 is merged, so this should be kept open until that is merged, right @jyu00 ?

**Comment by jyu00** [comment]
It was resolved by #455 when using the `Sampler` class.

**Comment by kevinsung** [comment]
Ah ok, this seems to still be an issue when loading jobs from the service but I guess that is the Result issue which is separate.
